subsys/mpsl/cx/thread/mpsl_cx_thread.c

```diff
@@ -11,7 +11,10 @@
  *
  */
 
+
+#if !defined(CONFIG_MPSL_CX_PIN_FORWARDER)
```

Comment:
I find confusing reusing the same module as "mpsl_cx_pin_forwarder" and "mpsl_cx_thread" implementation. The responsibility of the modules is different, only small part of the source file is shared, most is excluded by preprocessor macros. I think it would be cleaner to introduce a new module like `subsys/mpsl/cx/pin_forwarder.c` for this purpose.

Reply:
I don't agree. Consider you are customer, not developer of NCS. You want to add your custom implementation, howewer you pull NCS with west into your project. You don't want to have to tweak anything inside the NCS itself. You want to add another cx implementation. If pin_forwarder functionality is crammed into single "pin_forwarder.c" you will need to tweak this file to add your new cx plug-in. You cannot add new cx implementation seamlessly.
Let me remind you this PR https://github.com/nrfconnect/sdk-nrf/pull/6502

I think that this "pin-forwarding" feature is small enough that it can be kept within implementation.

Reply:
I disagree :)

While I'm creating my out-of-tree application for NCS I need an easy and clear way to define my boards and configuration for images for all cores. Also I would appreciate if creating new FEM or cx implementations was easy.

I would like pin forwarding to be handled by the system based on device tree only, without need for changes in any implementation like cx or FEM. If there was a generic pin-forwarder feature like I described in https://github.com/nrfconnect/sdk-nrf/pull/6706#discussion_r796387363, I would need to place appropriate `pin_forwarding` node in DTS file I prepare for my App Core image. And that's it. I would not need to change anything in any implementation related to coexistence or FEMs.

If I use any of the cx implementation provided in SDK, I simply add them to device tree for my Net Core. And that's it.

When I create a new cx implementation, I focus only on part related to coexistence itself, the one needed for Net Core. I do not want to implement pin forwarding in this file. With the patch I'm reviewing it's not the case. I would need to mix pin forwarding with new cx implementation I'm creating. I need to implement both pin forwarding feature and cx functionality in single source file, what is pretty hard to follow if you don't know SDK internals.

That's why I would suggest to separate pin forwarding from FEMs or cx implementations and make it explicit and easy to use.

Reply:
@hubertmis I like your proposal a lot, I think it's easy to use and elegant. The only problem I see with that approach is that I don't know yet how to implement it :) The `pin_forwarder` node would have to allow the user to provide an unspecified number of pins to forward and I'm not sure how to represent it in devicetree. I'll see what I can do.

Reply:
I'm pretty sure it's possible to iterate over child nodes in device tree. Let's dig into this to find the right solution!